### ISSUE TITLE
Refactor Classes to resolve dependency loops

### ISSUE THREAD

**Issue by csfreak** [rhtools/vminfo-parser#13]
Currently,  The `Analyzer`, `Visualizer`, and `CLIOuptut` classes are interdependent and have a dependency loop.    `Analyzer` and `main` both instantiate `Visualizer`,  and all three use various methods of access the `Config` and `VMData` object.

In order to cleanly separate them,  we need an agreed scope for each one,  and definitions of what logic should and should not belong to each.  While this may result in increase complexity in a few cases,  that increased complexity will result in easier to understand and maintain code.

I propose the following separations.

- [X]  `Analyzer` should hold all calculation logic.  The output of all methods in `Analyzer` should be some type of record(probably pandas DataFrame) with only the records needed to generate output.  For example,  if we are counting instances of each OS,  the output should have nothing besides the counts and names of each OS.  

- [X]  `CLIOutput` should be the only thing writing to stdout.  All other console output should be moved to logging (which should output to stderr for now). `CLIOutput` methods should receive only the relevant output from `Analyzer` and should return nothing.

- [X]  `Visualizer` should be the only module that calls Matplotlib lib, either directly or indirectly (via pandas plotting accessor). `Visualizer` methods should take as input only the relevant output from `Analyzer` and should return nothing.

- [x]  `Visualizer` should not be instantiated if `config.generate_graphs == False`.  

- [ ] #81

- [x]  If a consistent interface is created,  calling those methods can be delegated to `Analyzer`, but otherwise, `Analyzer` should not be concerned with output.

If these goals are accepted,  code changes to these modules should work towards these ends,  recognizing, that everything can't be changed at one time and PRs should still be small, digestible fragments.

**Comment by csfreak** [comment]
currently `Analyzer` calls `Visualizer` in both `handle_disk_space` and `handle_operating_system_counts`,  if the output modules had a consistent implementation,  `Analyzer` could instead hold a list of output modules that it called,  alternatively, it could just pass the data back to main and main could call them.

**Comment by stratus-ss** [comment]
I pulled visualizer apart as https://github.com/rhtools/vminfo-parser/pull/35

**Comment by stratus-ss** [comment]
CLIOutput was moved to its own module in https://github.com/rhtools/vminfo-parser/pull/39

**Comment by stratus-ss** [comment]
Analyzer moved to its own module in #56

**Comment by csfreak** [comment]
After #75, analyzer no longer handles any output.

**Comment by csfreak** [comment]
Only remaining item moved to new issue.
